freva-data-portal-worker/src/data_portal_worker/load_data.py

```diff
@@ -247,9 +243,12 @@ def load_object(self, key: str) -> LoadDict:
         KeyError: If the key doesn't exist in the cache (anymore).
         """
         data_cache = self.cache.get(key)
-        if data_cache is None:
+        dset_bin = self.cache.get(f"{key}-dset")
```

Comment:
and to make the error catching smarter, we can handle it via checking the metadata var in cache as well. i meant, if it has metadata and not dset, raise another error ... just popped up